src/main/java/fi/aalto/cs/apluscourses/model/Points.java

```diff
@@ -125,17 +126,12 @@ private static void parseSubmissions(@NotNull JSONObject exerciseJson,
 
   @Nullable
   private static Long parseSubmissionId(String submissionUrl) {
-    if (submissionUrl.isEmpty()) {
-      return null;
-    }
-    if (submissionUrl.endsWith("/")) {
-      submissionUrl = submissionUrl.substring(0, submissionUrl.length() - 1);
-    }
-    try {
-      return Long.parseLong(submissionUrl.substring(submissionUrl.lastIndexOf("/") + 1));
-    } catch (Exception e) {
+    var pattern = Pattern.compile("(\\d+)(?!.*\\d)");
```

Comment:
Am I correct, this takes the last continuous string of digits in the URL? I guess it works OK here, but would it make more sense to write the regex to reflect the expected form of the URL. Something like `"/submission/(\\d+)/?$"`. Or can URLs be of many different forms? You can also make the pattern object static final.

Reply:
I tried that, but it didn't work because I used matcher.group(0), but with matcher.group(1) it works. Thanks

Reply:
Yep, the `group(0)` is the whole matched string, `group(1)` is the substring in the first parantheses. 👍